### DISCUSSION TITLE
C&T F8680 emulation

### DISCUSSION THREAD

**Discussion by Torinde** [dbalsom/marty#139] [Ideas]
Interesting 186 variant with 26-bit addressing (64MB RAM): [datasheets](https://datasheets.chipdb.org/Chips/).
- uses bank switching for the space above 1MB?
- "maximum local system memory that is directly managed is 4MB" (page 20)... have to read more to understand what that refers to
- can it use [ISA card](https://github.com/86Box/86Box/discussions/4528)(s) for expanded memory in addition, for a total of 64+32=96MB RAM?

Has a System Management feature "SuperState R".

Kind of related is VAutomation [Turbo86/186](https://web.archive.org/web/20000930221809fw_/http://www.vautomation.com/turbo186/turbo.html) that has 24 bit addressing mode allowing for up to a 16Mbyte address space.

Related:
- https://github.com/joncampbell123/dosbox-x/discussions/5261, https://github.com/86Box/86Box/discussions/4922

**Comment by dbalsom** [comment]
Neat, but not going to happen unless someone wants to contribute it.

**Reply by Torinde** [reply]
I understand.

What is your opinion on how they implemented that 26-bit/64MB support? 
- other 86/186 CPUs which support 24-bit/16MB seem to use custom addressing modes (e.g. NEC V33)
- [EMS](https://en.wikipedia.org/wiki/Expanded_memory) maximum seems to be 32MB, but if you compare F8680 (rev1.1) vs F8680A (rev2.0) datasheets you'll see they omitted the text mentioning a 32MB restriction...
- at the same time both F8680 datasheets mention a 4MB "local directly managed" memory.

**Reply by dbalsom** [reply]
no idea. unfortunately, with limited time, i can't investigate every little rabbit hole in early PC history.
